From f7e7e3f51e18d0e275c69029f8781f1fb7932f43 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 3 Sep 2020 12:05:50 -0700
Subject: [PATCH 1561/1852] v4l: xilinx: sdirxss: Update the quantization

commit adb58e37795d3ce4b438631634b47e6830fc4781 from
https://github.com/Xilinx/linux-xlnx.git

Update the quantization for the v4l2_bus_framefmt based on the rules
mentioned in videodev2.h

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 0e03b7a920e7..8ac72785accf 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1360,6 +1360,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 	format->colorspace = V4L2_COLORSPACE_SMPTE170M;
 	format->xfer_func = V4L2_XFER_FUNC_709;
 	format->ycbcr_enc = V4L2_YCBCR_ENC_601;
+	format->quantization = V4L2_QUANTIZATION_LIM_RANGE;
 
 	if (mode != XSDIRX_MODE_SD_MASK) {
 		u8 eotf = (payload & XST352_BYTE2_EOTF_MASK) >>
@@ -1421,6 +1422,11 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 		}
 	}
 
+	/* Set quantization range */
+	if (sampling == XST352_BYTE3_COLOR_FORMAT_GBR &&
+	    format->colorspace != V4L2_COLORSPACE_BT2020)
+		format->quantization = V4L2_QUANTIZATION_FULL_RANGE;
+
 	dev_dbg(core->dev, "Stream width = %d height = %d Field = %d payload = 0x%08x ts = 0x%08x\n",
 		format->width, format->height, format->field, payload, val);
 	dev_dbg(core->dev, "frame rate numerator = %d denominator = %d\n",
-- 
2.31.1

